prov/efa/src/rxr/rxr_pkt_type_misc.c

```diff
@@ -63,45 +63,46 @@ ssize_t rxr_pkt_init_handshake(struct rxr_ep *ep,
 
 	pkt_entry->pkt_size = sizeof(struct rxr_handshake_hdr)
 			      + RXR_NUM_PROTOCOL_VERSION * sizeof(uint64_t);
-	pkt_entry->addr = addr;
+	pkt_entry->addr = rx_entry->addr;
+	pkt_entry->x_entry = rx_entry;
 	return 0;
 }
 
-void rxr_pkt_post_handshake(struct rxr_ep *ep,
-			    struct rxr_peer *peer,
-			    fi_addr_t addr)
+ssize_t rxr_pkt_post_handshake(struct rxr_ep *ep, struct rxr_peer *peer,
```

Comment:
Since you are here, please add a comment describe the function: input, output, behavior. etc